multichain-testing/test/cosmos-staking.test.ts

```diff
@@ -0,0 +1,94 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import type { TestFn } from 'ava';
+import { generateMnemonic } from '../tools/wallet.js';
+import { makeQueryClient } from '../tools/query.js';
+import { commonSetup } from './support.js';
+// import { makeDoOffer } from '../tools/e2e-tools.js';
+
+const test = anyTest as TestFn<Awaited<ReturnType<typeof commonSetup>>>;
+
+const KEYS = ['user1'];
+
+const deleteKeys = async (deleteKey: (name: string) => Promise<string>) => {
+  for (const key of KEYS) {
+    try {
+      await deleteKey(key);
+    } catch (_e) {
+      // ignore
+    }
+  }
+};
```

Comment:
You really want delete to fail silently?

```suggestion
const deleteKeys = (deleteKey: (name: string) => Promise<string>) => 
  Promise.all(KEYS.map(key => deleteKey(key).catch()));
```

Reply:
It was helpful when I was working out the correct pattern here. I've since refactored in `support.ts` to:
```ts
const makeKeyring: (e2eTools: Pick<E2ETools, "addKey" | "deleteKey">) => Promise<{
    setupTestKeys: (keys?: string[]) => Promise<Record<string, string>>;
    deleteTestKeys: () => Promise<string[]>;
}>
```
If a test needs keys it can call `setupTestKeys` in `test.before()` and `.deleteTestKeys()` in `test.after()`